cmd/checker/checker_test.go

```diff
@@ -0,0 +1,143 @@
+package checker
+
+import (
+	"kool-dev/kool/cmd/builder"
+	"errors"
+	"testing"
+)
+
+type FakeCommand struct {}
+
+func (c *FakeCommand) AppendArgs(args ...string) {}
+
+func (c *FakeCommand) LookPath() (err error) {
+	return
+}
+
+func (c *FakeCommand) Interactive() (err error) {
+	return
+}
+
+func (c *FakeCommand) Exec() (outStr string, err error) {
+	return
+}
+
+func (c *FakeCommand) String() (strCommand string) {
+	return
+}
+
+type NotInstalledDockerCmd struct {
+	FakeCommand
+}
+
+func (c *NotInstalledDockerCmd) LookPath() (err error) {
+	err = errors.New("not installed")
+	return
+}
+
+type NotRunningDockerCmd struct {
+	FakeCommand
+}
+
+func (c *NotRunningDockerCmd) Exec() (outStr string, err error) {
+	err = errors.New("not running")
+	outStr = "error"
+	return
+}
+
+type NotInstalledDockerComposeCmd struct {
+	FakeCommand
+}
+
+func (c *NotInstalledDockerComposeCmd) LookPath() (err error) {
+	err = errors.New("not installed")
+	return
+}
+
+func TestDefaultChecker(t *testing.T) {
+	var c Checker = NewChecker()
+
+	if _, assert := c.(*DefaultChecker); !assert {
+		t.Errorf("NewChecker() did not return a *DefaultChecker")
+	}
+}
+
+func TestDockerNotInstalled(t *testing.T) {
+	var c Checker
+
+	dockerCmd := &NotInstalledDockerCmd{}
+	dockerComposeCmd, _ := builder.ParseCommand("docker-compose ps")
+
+	c = &DefaultChecker{dockerCmd, dockerComposeCmd}
+
+	message, err := c.CheckKoolDependencies()
+
+	if err == nil {
+		t.Error("Expected an error, got none.")
+		return
+	}
+
+	if message != "Docker doesn't seem to be installed, install it first and retry." {
```

Comment:
By changing the Err* handling above suggested, here we could have only helper methods like `IsDockerNotFoundError(err)` which then implements the `err.Error() == ErrDockerNotInstalled.Error()`